src/kbs2.jl

```diff
@@ -68,7 +69,8 @@ function knuthbendix2!(rs::RewritingSystem, o::Ordering = ordering(rs))
     end
 
     i = 1
-    while i ≤ length(tmprs)
+    while i ≤ (ltmprs = length(tmprs))
+        ltmprs > maxrules && (println("Maximum number of rules in the RewritingSystem reached. You can try again with higher value."); break)
```

Comment:
no `print`ing of messages these go to stdout, while user may be watching a logger; use `@warn(msg)`